examples/google-sheet/toolpad/pages/page/page.yml

```diff
@@ -0,0 +1,94 @@
+apiVersion: v1
+kind: page
+spec:
+  id: mDJP5kU
+  title: Default page
+  queries:
+    - name: fetchSheet
+      query:
+        function: fetchSheet
+        kind: local
+      parameters:
+        - name: spreadsheetId
+          value:
+            $$jsExpression: |
+              select.value
+        - name: range
+          value:
+            $$jsExpression: |
+              textField.value
+    - name: fetchList
+      query:
+        function: functions.ts#fetchList
+        kind: local
+      parameters:
+        - name: spreadsheetId
+          value:
+            $$jsExpression: |
+              select.value
+        - name: range
+          value:
+            $$jsExpression: |
+              textField.value
+  content:
+    - component: PageRow
+      name: pageRow1
+      children:
+        - component: Select
+          name: select
+          props:
+            label: Choose Spreadsheet
+            fullWidth: true
+            options:
+              $$jsExpression: |
+                fetchList.data.map((a) => a.id)
+          layout:
+            columnSize: 1
+        - component: TextField
+          name: textField
+          props:
+            defaultValue: Sheet1!A:Z
+            label: Enter range
+            fullWidth: true
+    - component: PageRow
+      name: pageRow
+      children:
+        - component: DataGrid
+          name: dataGrid
+          props:
+            rows:
+              $$jsExpression: |
+                fetchSheet.data
+            columns:
```

Comment:
This won't work with the user's sheet, they would have different column names.

Reply:
So, I shared a new sheet with the service account and it doesn't pick column headers automatically. I am required to click datagrid and click on recreate column, which should be fine for now. 
It would be better if we could do it automatically or prompt the user to recreate columns.

Reply:
I'm opening https://github.com/mui/mui-toolpad/issues/2409